### PR TITLE
27 the card which was chosen is displayed in the users profile for the next hours

### PR DESCRIPTION
**Pull request by daveking7**
It doesn't look nice yet, the design is something somebody should do later

### CODE REVIEW

File: src/components/views/BlackCardSelection.js [hunk 1]
```diff
@@ -3,35 +3,63 @@ import "styles/views/GameMenu.scss";
...
+    // fetch the blackCards from the server (it is the server's responsibility to give us 8 cards)
     useEffect(() => {
         async function fetchCards() {
             try {
-                const response = await api.get(`games/${localStorage.getItem("id")}`);
+                const response = await api.get(`games/${id}`,
+                    {
+                        // reconfiguration might be necessary in case token is not in localStorage here
+                        headers: {
+                            "authorization": token
```
Comment (JoMuelle) [inline src/components/views/BlackCardSelection.js:43]: shouldn't the token be passed anyways in the api.js file? Or is this something different

File: src/components/views/Login.js [hunk 2]
```diff
@@ -56,7 +56,13 @@ const Login = () => {
       localStorage.setItem('id', response.data.id)
 
       // Login successfully worked --> navigate to the route /game in the GameRouter
-      history.push(`/game/select/blackCard`);
+      history.push({
+        pathname: `/game/select/blackCard`,
+        state: {
```
Comment (JoMuelle) [inline src/components/views/Login.js:61]: what does state: do?

**Review by JoMuelle — COMMENTED** [review]
Looks good.

File: src/components/views/BlackCardSelection.js (hunk 1 above)
Comment (TwoDigitsOneNumber) [reply src/components/views/BlackCardSelection.js:43]: not sure what you mean. having the auth token in the header of the request is standard practice.

**Review by TwoDigitsOneNumber — COMMENTED** [review]
This looks good to me, so i think we can merge it.
However, is it possible to get rid of some redundant code. i honestly can't see where there is much redundance but sonar cloud complains about this.

File: src/components/views/BlackCardSelection.js (hunk 1 above)
Comment (daveking7) [reply src/components/views/BlackCardSelection.js:43]: @JoMuelle Yes, but how we solved it introduced a bug. Will fix this in a seperate Issue
@TwoDigitsOneNumber We do not want to reconfigure the header each time we call api, so this is what he meant.

File: src/components/views/Login.js (hunk 2 above)
Comment (daveking7) [reply src/components/views/Login.js:61]: That passes a state (an object) with history to the next Component.

**Pull request merged**